test/src/e2e_vm_tests/test_programs/should_pass/language/configurable_consts_in_lib/src/lib.sw

```diff
@@ -0,0 +1,32 @@
+library lib;
+
+pub struct MyStruct {
+    x: u64, 
+    y: bool
+}
+
+pub enum MyEnum {
+    A: u64,
+    B: bool,
+}
+
+impl core::ops::Eq for MyEnum {
+    fn eq(self, other: MyEnum) -> bool {
+        match (self, other) {
+            (MyEnum::A(inner1), MyEnum::A(inner2))  => inner1 == inner2,
+            (MyEnum::B(inner1), MyEnum::B(inner2))  => inner1 == inner2,
+            _ => false,
+        }
+    }
+}
+
+configurable {
+    C0: bool = true,
+    C1: u64 = 42,
+    C2: b256 = 0x1111111111111111111111111111111111111111111111111111111111111111,
+    C3: MyStruct = MyStruct { x: 42, y: true },
+    C4: MyEnum = MyEnum::A(42),
+    C5: MyEnum = MyEnum::B(true),
+    C6: str[4] = "fuel",
+    C7: [u64; 4] = [1, 2, 3, 4],
+}
```

Comment:
Perhaps we should be a little conservative here and disallow `configurable` blocks in `library`s? I think `configurable` in libraries raises many of the same questions as we have in #2585.

E.g. what if a library with a `configurable` block is depended on by two other library dependencies? Does it show up in the data section once for each time its depended on? Or only once altogether? How should `forc` handle `configurable` sections in transitive dependencies? We have ideas [here](https://github.com/FuelLabs/sway-rfcs/pull/19#issuecomment-1286351753) on how to handle top-level dependencies, but it's not yet clear how we'd handle transitive deps.

I think only allowing configurable in scripts, predicates and contracts for now is probably safe - it covers the only motivating use-case we came up with in RFC 19 ([contract factories](https://github.com/FuelLabs/sway-rfcs/pull/19/files#diff-47c0979605623f4b3f39c63e1b462abe798abcb7a775e833d4e91817f1c2bbe3R20)).

Reply:
Thank you @mitchmindtree, this makes sense and I'm happy to disallow this for now. You're right that we can cover the required use cases without libraries - I was strictly thinking of top level dependencies here and I suspect this PR disallows problematic transitive dependencies via the global check I have in program.rs, but I haven't tested that yet so I could be wrong 😆. Either way, I will disallow libraries for now and we can handle top level dependencies separately as part of the `forc` side of this feature.

Reply:
On that note, I think I will wrap up the RFC before marking this PR as ready for review. I won't make much progress this week though as I'm mostly ooo.